pkg/codec/vaapi/vp8.go

```diff
@@ -107,17 +107,32 @@ type encoderVP8 struct {
 
 // newVP8Encoder creates new VP8 encoder
 func newVP8Encoder(r video.Reader, p prop.Media, params ParamsVP8) (codec.ReadCloser, error) {
-	if p.Width%16 != 0 || p.Width == 0 {
+	var (
+		width, height int
+		frameRate     float32
+		ok            bool
+	)
+	if width, ok = p.Width.Value(); !ok {
+		return nil, errors.New("media width is not constrained")
+	}
+	if height, ok = p.Height.Value(); !ok {
+		return nil, errors.New("media height is not constrained")
+	}
+	if frameRate, ok = p.FrameRate.Value(); !ok {
+		return nil, errors.New("media frame rate is not constrained")
+	}
```

Comment:
Maybe better to separate `prop.Media` (exact values) and `prop.MediaConstraint` (constraints)